src/main/java/dev/shivamnagpal/users/dtos/response/wrapper/ResponseWrapper.java

```diff
@@ -0,0 +1,43 @@
+package dev.shivamnagpal.users.dtos.response.wrapper;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import dev.shivamnagpal.users.enums.ResponseStatus;
+
+import java.time.Instant;
+import java.util.List;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public record ResponseWrapper<T>(
+        T data, PaginationResponse pagination, List<ErrorResponse> errors,
+        MetaResponse metaResponse
+) {
+
+    public static <T> ResponseWrapper<T> success(T data) {
+        return success(data, null);
+    }
+
+    public static <T> ResponseWrapper<T> success(T data, PaginationResponse pagination) {
+        MetaResponse metaResponse = MetaResponse.builder()
+                .status(ResponseStatus.SUCCESS)
+                .timestamp(Instant.now().toEpochMilli())
+                .build();
+        return new ResponseWrapper<>(data, pagination, null, metaResponse);
+    }
+
+    public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errorResponses) {
+        MetaResponse metaResponse = MetaResponse.builder()
+                .status(ResponseStatus.FAILURE)
+                .timestamp(Instant.now().toEpochMilli())
+                .build();
+        return new ResponseWrapper<>(null, null, errorResponses, metaResponse);
```

Comment:
Just call the bigger overload method here
`return failure(errorResponses, null)`

Reply:
fixed